lib/driver.rb

```diff
@@ -0,0 +1,80 @@
+require_relative 'csv_record'
+
+module RideShare
+  class Driver < CsvRecord
+    attr_reader :id, :name, :vin, :status, :trips
+
+    def initialize(id:, name:, vin:, status: :AVAILABLE, trips: [] )
+      valid_driver_status = [:UNAVAILABLE , :AVAILABLE]
+
+      super(id) #calling on the super class csv_record
+
+      @name = name
+      @vin = vin
+      raise ArgumentError, 'String must be length 17' if vin.length != 17
+      @status = status
+      raise ArgumentError, 'Driver status must be available or unavailable' unless valid_driver_status.include?(status)
+      @trips = trips
+
+    end
+
+    def add_trip(trip)
+      @trips << trip
+    end
+
+    def average_rating
+      if trips.empty?
+        average_rating = 0
+      else
+        completed_trips = @trips.select do |trip|
+          trip.rating != nil
+        end
+        total_ratings = completed_trips.sum do |trip|
+           trip.rating
+        end
+
+        average_rating = total_ratings.to_f/completed_trips.length
+      end
+
+      return average_rating
+    end
+
+
+    def total_revenue
+      total_revenue = 0.0
+
+      if @trips.empty?
+        return total_revenue
+      else
+        completed_trips = @trips.select do |trip|
+          trip.cost != nil
+        end
+        completed_trips.each do |trip|
+          if trip.cost < 1.65
+            total_revenue += trip.cost
+          elsif trip.cost >= 1.65
+            total_revenue += (0.80 * (trip.cost - 1.65))
+          end
+        end
+      end
+
+      return total_revenue.round(2)
+    end
+
+    def update_status(trip_in_progress)
```

Comment:
I love that this is a method defined on the Driver rather than putting this code directly in the TripDispacter :)